cores/esp8266/Schedule.cpp

```diff
@@ -76,88 +80,110 @@ bool schedule_function_us(std::function<bool(void)>&& fn, uint32_t repeat_us, sc
     return true;
 }
 
-IRAM_ATTR // (not only) called from ISR
-bool schedule_function_us(const std::function<bool(void)>& fn, uint32_t repeat_us, schedule_e policy)
+bool schedule_recurrent_function_us (const std::function<bool(void)>& fn, uint32_t repeat_us)
 {
-    return schedule_function_us(std::function<bool(void)>(fn), repeat_us, policy);
-}
+    assert(repeat_us < decltype(recurrent_fn_t::callNow)::neverExpires); //~26800000us (26.8s)
 
-IRAM_ATTR // called from ISR
-bool schedule_function(std::function<void(void)>&& fn, schedule_e policy)
-{
-    return schedule_function_us([fn]() { fn(); return false; }, 0, policy);
+    esp8266::InterruptLock lockAllInterruptsInThisScope;
+
+    recurrent_fn_t* item = new recurrent_fn_t(repeat_us);
```

Comment:
@d-a-v Elsewhere you changed new to malloc, but here new is still used. Is that on purpose?

Reply:
Yes, on purpose, because `schedule_recurrent_function_us()` is never called from sys context.